### ISSUE TITLE
[Bug]: NVDA still reads combo box as editable even when allowFreeform is set to false

### ISSUE THREAD

**Issue by jepetty** [microsoft/fluentui#33049]
### Component

Combobox

### Package version

^9.0.0

### React version

^17

### Environment

Note: this is from my local since I couldn't figure out how to get from sandbox so some information is contradictory from what was above
  System:
    OS: Windows 11 10.0.26100
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 22.43 GB / 63.95 GB
  Browsers:
    Edge: Chromium (129.0.2792.89)
    Internet Explorer: 11.0.26100.1
  npmPackages:
    @fluentui/react: ^8.110.1 => 8.110.1
    @fluentui/react-components: ^9.20.4 => 9.20.4
    @types/react: ^16.14.34 => 16.14.34
    @types/react-dom: ^16.9.17 => 16.9.17
    react: ^16.14.0 => 16.14.0
    react-dom: ^16.14.0 => 16.14.0

### Current Behavior

Hi! My team has an accessibility bug where screenreaders (NVDA, JAWS) are announcing non-editable combo boxes as editable. I've tried a bunch of different things, though from documentation it sounds like setting `allowFreeform` to false should be enough. Is this true? It seems even with this set, screenreaders still announce editable
![Image](https://github.com/user-attachments/assets/ba5405f3-e0e5-459e-a59f-3602f75c090c)

### Expected Behavior

I am expecting that `allowFreeform` set to false should result in the combo box not being announced as editable. Is there another way to achieve this?

### Reproduction

https://codesandbox.io/p/sandbox/stoic-bird-79h6rp?workspaceId=cd9e937e-b92d-4a50-950a-dca9e2ef7172

### Steps to reproduce

This reproduction is super easy!
1. Download a screenreader
2. Edit any of the samples on the website to add `allowFreeform={false}` to the combo box
3. Observe screenreader still lists combo box as editable

### Are you reporting an Accessibility issue?

None

### Suggested severity

High - No workaround

### Are you willing to submit a PR to fix?

yes

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by jepetty** [comment]
I said I'm open to fix, but will need pointers as I've never worked in this code base before :D

**Comment by smhigley** [comment]
NVDA & JAWS are correct here -- that is editable, even if it doesn't allow freeform input. It's editable if you can type text into it, which you can. The value change on blur is a separate thing, and should not impact whether screen reader users are made aware that they're in an edit box.

The truly non-editable control would be a Dropdown, not a Combobox -- if you don't want users to be able to type/edit, that would be the control you want 😊.

Hope that helps!

**Comment by smhigley** [comment]
Also @jepetty even though I closed this, please feel free to reach out directly if you have any other questions on screen reader + combobox behavior, or re-open if you feel like I missed something about the interaction you're trying to create.
